overlord/snapstate/backend_test.go

```diff
@@ -798,24 +800,29 @@ func (f *fakeSnappyBackend) CopySnapData(newInfo, oldInfo *snap.Info, p progress
 	return nil
 }
 
-func (f *fakeSnappyBackend) LinkSnap(info *snap.Info, model *asserts.Model, tm timings.Measurer) error {
+func (f *fakeSnappyBackend) LinkSnap(info *snap.Info, model *asserts.Model, disabledSvcs []string, tm timings.Measurer) error {
 	if info.MountDir() == f.linkSnapWaitTrigger {
 		f.linkSnapWaitCh <- 1
 		<-f.linkSnapWaitCh
 	}
 
+	op := fakeOp{
+		op:   "link-snap",
+		path: info.MountDir(),
+	}
+
+	// only add the services to the op if there's something to add
+	if len(disabledSvcs) != 0 {
+		op.services = disabledSvcs
```

Comment:
It is slightly confusing that we keep disabled services in 'services' member; should we have a new member for them?

Reply:
Sure, I'll add them into a different member

Reply:
Adding this revealed that my test coverage of the methods on the fakeSnappyBackend were lacking as none of them failed after initially renaming this, so I'm adding a couple more tests that exercise this part, thanks for the suggestion!

Reply:
This should be done now